web-api/src/embedding.rs

```diff
@@ -17,14 +17,26 @@ use xayn_ai_bert::{AvgEmbedder, Config as EmbedderConfig, NormalizedEmbedding};
 
 use crate::{app::SetupError, error::common::InternalError, utils::RelativePathBuf};
 
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(rename_all = "lowercase")]
+pub enum Config {
```

Comment:
:art: we could use an untagged enum for serde, then the current serialized format would stay the same and it wouldn't introduce another nested table in the toml. (this of course only works if `Pipeline` and the new sagemaker config object don't have the exact same fields, which they probably don't have.)
```suggestion
#[serde(rename_all = "lowercase", untagged)]
pub enum Config {
```

Reply:
untagged is a problem because the user can put the field of both and then the behaviour depends on the order of the enum. I don't know if untagged + deny_unknow can help here, but I prefer internally tagged but there actually the problem of backward compatibility, I don't know if we can specify a default for the tag if not present.

Reply:
that's true. with the current change to use the internally tagged enum the nesting is gone, so that should be fine.